config.js

```diff
@@ -415,7 +412,51 @@ if (argv.botid || argv.bot || argv.id || argv.minrankedhandicap || argv.maxranke
         }
     }
 
-    if (argv.maxrank) {
+    if (argv.minrankranked) {
+        let re = /(\d+)([kdp])/;
+        let results = argv.minrank.toLowerCase().match(re);
+
+        if (results) {
+            if (results[2] == "k") {
+                exports.minrankranked = 30 - parseInt(results[1]);
+            } else if (results[2] == "d") {
+                exports.minrankranked = 30 - 1 + parseInt(results[1]);
+            } else if (results[2] == "p") {
+                exports.minrankranked = 36 + parseInt(results[1]);
+                exports.proonly = true;
+            } else {
+                console.error("Invalid minrankranked " + argv.minrankranked);
+                process.exit();
+            }
+        } else {
+            console.error("Could not parse minrankranked " + argv.minrankranked);
+            process.exit();
+        }
```

Comment:
Consider making a function of the rank parse code.
```
if (argv.minrankranked) {
  try {
  [exports.minrankranked, exports.proonly] = parseRank(argv.minrank);
  } catch (e) {
    if (e === Invalid) {
        console.error("Invalid minrankranked " + argv.minrankranked);
    } else {
        console.error("Could not parse minrankranked " + argv.minrankranked);
    }
    process.exit();
  }
}
```

```
parseRank(rank) {
    let re = /(\d+)([kdp])/;
    let results = argv.minrank.toLowerCase().match(re);
    if (results) {
        if (results[2] == "k") {
            return [30 - parseInt(results[1]), false];
        } else if (results[2] == "d") {
            return [30 - 1 + parseInt(results[1]), false];
        } else if (results[2] == "p") {
            return [36 + parseInt(results[1]), true];
        } else {
            throw "Invalid";
        }
    } else {
        throw "Parse";
    }
}
```

Reply:
I wrote this code really fast, so i hope it works. Please test. If it doesnt work, let me know or just fix it yourself :D

Reply:
same as other comments, i'm really thankful for all these awesome code optimizations, and i'm looking forward to try them and implement them if they work !!

but !!!!!

there are still tons of things to first or add first :rofl: 

just to add the boardsize to human readable function, i got to fix 3 different things (undefined error, then always reject, then reject message)

i was willing to go through this because there are direct benefits for the user :smile: 

but in this case we would want to rework all the code at once in a anti repetition code, so i'd rather consider this when we do it for all parts of the code that have same issue (spring cleaning :laughing: !!), and also when i have more knowledge to understand whats going on and fix it :100: